src/Compilers/Core/Portable/Syntax/ChildSyntaxList.cs

```diff
@@ -83,17 +83,46 @@ private static int Occupancy(GreenNode green)
             return green.IsList ? green.SlotCount : 1;
         }
 
+        internal struct SlotData(int slotIndex, int precedingOccupantSlotCount, int positionAtSlotIndex)
```

Comment:
>internal struct [](http://example.com/codeflow?start=8&length=15)

Consider using a `readonly struct`.

Reply:
I don't think that will work as these values are changed in the ItermInternal* methods to narrow the search on subsequent calls.

Reply:
It looks like all three fields are set in each case, so we could replace the entire struct value instead.

Reply:
so, instead of being a ref parameter to the method, it would need to be taken in and returned separately?

Reply:
It could be kept as a `ref` parameter, and the entire value replaced in the method:
```csharp
if (slotIndex != slotData.SlotIndex)
{
    slotData = new SlotData(slotIndex, index - idx, position);
}
```

Reply:
Duh Todd! Thanks for sticking with me through that! :)